cpp/ql/src/semmle/code/cpp/rangeanalysis/SimpleRangeAnalysis.qll

```diff
@@ -1174,13 +1170,13 @@ private float getPhiUpperBounds(StackVariable v, RangeSsaDefinition phi) {
   or
   exists(VariableAccess access, float neConstant, float upper |
     isNEPhi(v, phi, access, neConstant) and
-    upper = getFullyConvertedUpperBounds(access) and
+    upper = getTruncatedUpperBounds(access) and
```

Comment:
This change is required because in C++ when we write `if (myInt)` there will be a `BoolConversion` on `access`, and so the fully-converted upper bounds will be at most 1.

I thought `getTruncatedUpperBounds` would be right for the `upperBoundFromGuard` case above too, but if I make that change then I break the tests that this PR adds in `two_bounds_from_one_test`. No other tests break, so our coverage was lacking before.